presto-spi/src/main/java/com/facebook/presto/spi/statistics/ExternalPlanStatisticsProvider.java

```diff
@@ -13,17 +13,32 @@
  */
 package com.facebook.presto.spi.statistics;
 
+import com.facebook.presto.common.plan.PlanCanonicalizationStrategy;
 import com.facebook.presto.spi.plan.PlanNode;
 import com.facebook.presto.spi.plan.TableScanNode;
 
+import java.util.Map;
 import java.util.function.Function;
 
 public interface ExternalPlanStatisticsProvider
 {
     String getName();
 
-    PlanStatistics getStats(
-            PlanNode plan,
+    /**
+     * Get historical statistics for given plan nodes.
+     * Depending on `useHashBasedStatistics`, implementation may choose to use plan node hashes supplied,
```

Comment:
nit: when writing javadoc, try use the existing syntax like `{@param useHashBasedStatistics}`